scripts/docblock.mjs

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env zx
+
+/*
+ * Copyright 2024 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { parseFileSync } from '@swc/core';
+import { parse } from 'comment-parser';
+import { fs, glob } from 'zx';
+
+const SWC_OPTIONS = {
+  syntax: 'typescript',
+  target: 'es2022',
+};
+
+function exportsReducer(acc, node) {
+  if (node.type === 'ExportDeclaration') {
+    if (node.declaration.identifier) {
+      acc.push(node.declaration.identifier.value);
+    } else if (node.declaration.declarations) {
+      for (const inner of node.declaration.declarations) {
+        acc.push(inner.id.value);
+      }
+    }
+  }
+
+  return acc;
+}
+
+function getFileDetails(path) {
+  const ast = parseFileSync(path, SWC_OPTIONS);
+  const source = fs.readFileSync(path, 'utf8');
+
+  const exports = ast.body.reduce(exportsReducer, []);
+  const [pragma, ...list] = pragmaParser(source);
+
+  const result = [source];
+
+  switch (pragma) {
+    case 'ignore':
+      result.push(
+        list[0] === '*' ? [] : exports.filter((name) => !list.includes(name)),
+      );
+      break;
+    case 'only':
+      result.push(exports.filter((name) => list.includes(name)));
+      break;
+    default:
+      result.push(exports);
+      break;
+  }
+
+  return result;
+}
+
+const pragmaParser = (() => {
```

Comment:
Readability nit: Can we get this fn above `getFileDetails` since that is the calling order

Reply:
I try and rely on objective ordering this is in alpha order rather than anything subjective. I can move it around if necessary though.

Reply:
Every human reads top-to-bottom so it makes it less work (scroll up and down, "was that an import", scroll up, "nope, where is it defined then", scroll down, back and forth) to understand what is going on in examples like this. Where it is basically procedural. That being said, content like constant files, types files or function files with no internal dependencies I don't have an issue with them being alpha order.

Reply:
I broke them out into separate modules to make it cleaner overall; and potentially we could reuse pieces.